hubble-proxy/Makefile

```diff
@@ -0,0 +1,24 @@
+TARGET=hubble-proxy
+GIT_BRANCH != which git >/dev/null 2>&1 && git rev-parse --abbrev-ref HEAD
+GIT_HASH != which git >/dev/null 2>&1 && git rev-parse --short HEAD
```

Comment:
I wonder if it would make sense to use Cilium's versioning infrastructure here instead:
https://github.com/cilium/cilium/blob/master/Makefile.defs#L34

I think we probably also want to pull in all the other Makefile variables from `../Makefile.defs` so we end up building all binaries with the same `$GOBUILD` etc. See for example the Makefile of the Cilium CLI: https://github.com/cilium/cilium/blob/6d29bba62d7cef9d340848d360da2effc09f2485/cilium/Makefile

Reply:
I thought about this as well but in the end, although `hubble-proxy` lives in `cilium`'s repository (primarily to make integration testing easier), it's entirely independent from `cilium` as it will only interact with the hubble API and will run in an entirely separate container and even pod.
I'm also OK to make use of variables defined in `../Makefile.defs` but I'm not convinced yet that it's the better way to do it.

Reply:
Seems like there's two questions here:

* Should hubble-proxy use Cilium `Makefile` infrastructure?

My gut feeling here is that creating a new module, developers naturally want to create the simplest possible `Makefile`. Keeps things simple and they just work and it's easier to understand. Over time you need more and more from the make infrastructure that eventually motivates just consuming the same buildchain as the rest of the project. If this module lives in the Cilium project long-term, I would expect this step inevitable; you either reimplement the wheel or reuse it. There's a range of common functionality like avoiding pitfalls around shell invocation by setting the right shell options, to allowing users to configure their container runtime CLI / Go binary to consuming the same `$GOBUILD` flags as the rest of the build when running/testing. Maybe it's not needed today but I expect it's necessary one day.

* What's the intended semantic version handling for `hubble-proxy`?

For this, we can either tie it into Cilium semver or not. Seems like your answer here is expressing desire to allow a separate release / distribution flow for hubble-proxy, so detaching the semver would help with this. Just be aware that if this tool is sitting in the tree primarily for easier integration testing.. the version that will be tested there will be Cilium in the tree + hubble-proxy in the tree; compatibility with other versions, API breakage, etc. will be up to developers to manage and figure out how to continuously integrate.

Reply:
>     * Should hubble-proxy use Cilium `Makefile` infrastructure?

For the time being, I don't think it's necessary. By including `Makefile.defs`, this also defines variables that are non-relevant or even confusing for `hubble-proxy` (eg: `VERSION` as the `hubble-proxy` version may not be tied to the cilium version).

>     * What's the intended semantic version handling for `hubble-proxy`?
> 
> For this, we can either tie it into Cilium semver or not. Seems like your answer here is expressing desire to allow a separate release / distribution flow for hubble-proxy, so detaching the semver would help with this. Just be aware that if this tool is sitting in the tree primarily for easier integration testing.. the version that will be tested there will be Cilium in the tree + hubble-proxy in the tree; compatibility with other versions, API breakage, etc. will be up to developers to manage and figure out how to continuously integrate.

For the time being, this is not defined. Hubble proxy releases could be made independently of cilium as long as hubble's grpc interface remains compatible.

Reply:
The proxy releases (github tags) can not be made separate from cilium, as the trees are tied together. We can push quay images, but that's about it.

Reply:
Unless there are blockers from us using the existing makefiles, I'd get that straight from the start. Hubble and cilium often live in two completely separate worlds with their own structure due to the team allocation. It makes sense to bridge that a little bit given the current interweave between the two

Reply:
Update the PR to use cilium's make definitions as we as version.